project/src/main/steam/unique-level-achievement.gd

```diff
@@ -13,18 +13,24 @@ extends SteamAchievement
 ## The level ID whose achievement condition should be checked.
 export (String) var level_id: String
 
+## 'true' if the prepare_level_listeners() method has already been called
+var _prepared_level_listeners := false
+
 func connect_signals() -> void:
 	.connect_signals()
 	disconnect_save_signal()
 	disconnect_load_signal()
-	Breadcrumb.connect("after_scene_changed", self, "_on_Breadcrumb_after_scene_changed")
+	PuzzleState.connect("game_prepared", self, "_on_PuzzleState_game_prepared")
 
 
 ## Overridden by child classes to connect listeners to the Puzzle or PuzzleState.
 func prepare_level_listeners() -> void:
 	pass
 
 
-func _on_Breadcrumb_after_scene_changed() -> void:
-	if get_tree().current_scene is Puzzle and CurrentLevel.level_id == level_id:
+func _on_PuzzleState_game_prepared() -> void:
+	if get_tree().current_scene is Puzzle \
+			and CurrentLevel.level_id == level_id \
+			and not _prepared_level_listeners:
 		prepare_level_listeners()
+		_prepared_level_listeners = true
```

Comment:
Need to set `_prepared_level_listeners` to false at some point, or signals will not be reconnected the second time.

Reply:
Added a `Breadcrumb.connect("before_scene_changed"` which sets this to false.